MACHINE-LEARNING-SERVICE/src/models/objectML.js

```diff
@@ -14,30 +14,47 @@ const tf = require('@tensorflow/tfjs-node');
 const fs = require('fs').promises;
 
 // Represents the object recognition system
-class ObjectML{
-
-  constructor (image, percentage, objectRequired){
-    this.image = image;
+class ObjectML {
+  constructor(pathFile, percentage, objectRequired) {
+    this.pathFile = pathFile;
     this.percentage = percentage;
     this.objectRequired = objectRequired;
   }
- 
+
   // Allows to load the model and decode the image in order to make a detection of the desired object.
-  async predict(){
-    const results = await Promise.all([cocoSsd.load(), fs.readFile(this.image)]);
-    const model = results[0];
-    const imgTensor = tf.node.decodeImage(new Uint8Array(results[1]), 3);
-    const predictions = await model.detect(imgTensor);
-    const arrayObjects = [];
-    predictions.every((prediction)=> {
-      if(this.objectRequired == prediction.class && prediction.score >= this.percentage){
-        arrayObjects.push({'object': prediction.class, 'Score': prediction.score});
-        return false;
-      } else{
-        return true;
-      }
+  async predict() {
+    const model = await cocoSsd.load();
+    const imagesArray = await fs.readdir(this.pathFile);
+
+    const imagesToPredictArray = await Promise.all(
+      imagesArray.map(async (fileName) => {
+        const img = await fs.readFile(`${this.pathFile}${fileName}`);
+        const imgTensor = tf.node.decodeImage(new Uint8Array(img), 3);
+        const predict = await model.detect(imgTensor);
+        const data = { predict, fileName };
+        return data;
+      })
+    );
+
+    const foundObjectsArray = [];
+    imagesToPredictArray.forEach((predictions) => {
```

Comment:
I realize you are looking for the objects that are equal to object name and acceptance percent (from 40 to 55 line).
You could do that while you are iterating in line 30 using map. After line 34, you could verify with an if()

Reply:
I need to do both bucles, in the first one I need all the predict results and then in the second one I need to use .forEach because I don't want all the results.